### DISCUSSION TITLE
Working with float32

### DISCUSSION THREAD

**Discussion by shayakhmetovi** [FreeOpcUa/opcua-asyncio#1688] [Q&A] [answered]
When reading float32 from a third-party opcua server, the data is converted to float64. On the server 34.535. 
![float](https://github.com/user-attachments/assets/e6924957-be55-4225-9042-df2319f8341f)

When reading with the asyncua client, we get 34.53499984741211
![image](https://github.com/user-attachments/assets/43c35ad1-4fda-4041-9910-4d918255e85b)

How to work with float32 correctly?

asyncua 1.1.5
python 3.8

**Comment by oroulet** [comment, marked as answer]
Python has only int and float so whatever you get from ua is converted to that you cannot do anything as far as I know
